gradle-plugin/plugin/src/main/kotlin/com/emergetools/android/gradle/tasks/internal/SaveExtensionConfigTask.kt

```diff
@@ -25,24 +28,24 @@ import java.io.File
  * as a workaround to write the extension to JSON, where we can later assert on the
  * contents of the JSON file to ensure our config is handled properly.
  */
+@DisableCachingByDefault(because = "This task does not declare any outputs.")
 abstract class SaveExtensionConfigTask : DefaultTask() {
   @get:Input
   abstract val emergePluginExtension: Property<EmergePluginExtension>
 
-  private var outputPath: String? = null
-
+  @Input
   @Option(option = "outputPath", description = "Output path for configuration JSON file")
-  fun setOutputPath(outputPath: String) {
-    check(outputPath.isNotBlank()) { "Output path cannot be blank" }
-    this.outputPath = outputPath
-  }
+  abstract fun getOutputPath(): Property<String>
+
+  private val outputFile: Provider<RegularFile>
```

Comment:
Technically this should be the output and the task should be cacheable but enabling caching caused the task serialization to crash. Will need to examine this further in a separate PR.

Reply:
This task is also just for functionalTest so we can verify the config at runtime. Won't actually be used by any real users. There's likely a better way to do this as well, we can chat more about the motivation of this.